packages/strapi-plugin-users-permissions/admin/src/containers/Providers/index.js

```diff
@@ -30,6 +30,7 @@ const ProvidersPage = () => {
   const buttonSubmitRef = useRef(null);
   const [showForm, setShowForm] = useState(false);
   const [providerToEditName, setProviderToEditName] = useState(null);
+  const [providerIsWithSubdomain, setProviderIsWithSubdomain] = useState(false);
```

Comment:
The useState hook is not needed here you can replace it by the following:


```

import { get, has, upperFirst } from 'lodash';
const isProviderWithSubdomain = useMemo(() => {
    if (!providerToEditName) {
      return false;
    }

    const providerToEdit = providers.find(obj => obj.name === providerToEditName);

    return has(providerToEdit, 'subdomain');
  }, [providers, providerToEditName]);
```


Then you will just need to update the formToRender constant:

```
const formToRender = useMemo(() => {
  if (providerToEditName === 'email') {
    return forms.email;
  }

  if (isProviderWithSubdomain) {
    return forms.providersWithSubdomain;
  }

  return forms.providers;
}, [providerToEditName, isProviderWithSubdomain]);
```

Reply:
for the `return has(...)` I had to add     return has(providerToEdit, 'subdomain') && providerToEdit.subdomain !== undefined; as the previous code in the createProviderArray is adding the subdomain property to all providers.

Reply:
I am sorry I have forgotten to remove my comment on the createProvidersArray file if you keep your old code (really sorry about that) the has condition should work.

Reply:
@soupette rolled back to my previous code